tests/functional/adapter/dremio_specific/test_reflections.py

```
@@ -170,6 +190,8 @@ def models(self):
             "default_displays_model.sql": default_displays_model,
             "name_reflection_from_alias.sql": name_reflection_from_alias_model,
             "name_reflection_from_filename.sql": name_reflection_from_filename_model,
+            "wait_strategy_timeout_reflection.sql": wait_strategy_timeout_reflection,
+            "trigger_strategy_timeout_reflection.sql": trigger_strategy_timeout_reflection,
```

I realize testing `depend` is not that easy, but there must be a way. Is it possible to create a reflection that will never be finish materializing and then try to depend on it?

---

The idea of `depend` is that it will be stuck at the job running that reflection until it is complete. Even if we could make a reflection that would never materialize we'd just be stuck waiting for it forever

---

My idea would've been to assert that the function hasn't returned after a few seconds and then killing the thread. But I don't know how complex we want this to get

---

Yes, I agree with Simon that we should have guarantees that it will end. Waiting without a proof for a green result may be frustrating for customers.

---

That would be the `wait` strategy where the customer would set the maximum time they'd want to wait. Should we remove `depend` in that case? Or have it act as `wait` but throw an error instead of just skipping?